firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/KeyValueBuilder.kt

```diff
@@ -16,24 +16,46 @@
 
 package com.google.firebase.crashlytics
 
-/** Helper class to enable fluent syntax in [setCustomKeys] */
-class KeyValueBuilder(private val crashlytics: FirebaseCrashlytics) {
+/** Helper class to enable convenient syntax in [setCustomKeys] */
+class KeyValueBuilder
+private constructor(
+  private val crashlytics: FirebaseCrashlytics?,
+  private val builder: CustomKeysAndValues.Builder,
+) {
+  @Deprecated("Do not construct this directly. Use `setCustomKeys` instead. To be removed in 2025.")
```

Comment:
Maybe word it as "To be removed in the next major release"

Reply:
Done